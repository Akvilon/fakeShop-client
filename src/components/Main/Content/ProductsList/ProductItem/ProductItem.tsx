import React from 'react'
import {Button, ButtonGroup, Card, CardBody, CardFooter, Divider, Heading, Stack, Image} from "@chakra-ui/react";
import { Product } from '../../../../../redux/features/products/types'

type ProductItemType = {
   product: Product
}

export const ProductItem: React.FC<ProductItemType> = ({product}) => {
    return (
        <>
            <Card width='380px' data-testid="product-item">
                <CardBody>
                    <Image
                        src={product.images[2]}
                        alt=''
                        borderRadius='lg'
                    />
                    <Stack mt='6' spacing='3'>
                        <Heading size='md'>{product.title}</Heading>
                    </Stack>
                </CardBody>
                <Divider />
                <CardFooter>
                    <ButtonGroup spacing='2'>
                        <Button variant='solid' background="#319795" color="#fff">
                            Buy now
                        </Button>
                        <Button variant='ghost' color="#319795">
                            Add to cart
                        </Button>
                    </ButtonGroup>
                </CardFooter>
            </Card>
        </>
    )
}