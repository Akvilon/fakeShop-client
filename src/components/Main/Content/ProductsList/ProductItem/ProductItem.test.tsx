import { ProductItem } from './ProductItem'
import { render, screen } from '@testing-library/react'

describe('ProductItem component', () => {
    const product = {
        _id: 1,
        title: 'title',
        price: 10,
        description: 'description',
        images: ['','',''],
        category: {
            id: 1,
            name: 'category name',
            image: 'image'
        }
    };

    it('should match snapshot', () => {
        const { asFragment } = render(<ProductItem product={product} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('should render the "Buy now" button with the correct text', () => {
        render(<ProductItem product={product} />);
        const buyButton = screen.getByText(/buy now/i);
        expect(buyButton).toBeInTheDocument();
    });

    it('should render the "Add to cart" button with the correct text', () => {
        render(<ProductItem product={product} />);
        const addButton = screen.getByText(/add to cart/i);
        expect(addButton).toBeInTheDocument();
    });
});