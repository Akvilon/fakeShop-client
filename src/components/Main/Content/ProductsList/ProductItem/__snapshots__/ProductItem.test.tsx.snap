// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductItem component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="chakra-card css-1uci29l"
    data-testid="product-item"
  >
    <div
      class="chakra-card__body css-0"
    >
      <img
        alt=""
        class="chakra-image css-j17amb"
        src=""
      />
      <div
        class="chakra-stack css-10aolb9"
      >
        <h2
          class="chakra-heading css-0"
        >
          title
        </h2>
      </div>
    </div>
    <hr
      aria-orientation="horizontal"
      class="chakra-divider css-1upb9tn"
    />
    <div
      class="chakra-card__footer css-k008qs"
    >
      <div
        class="chakra-button__group css-1mlmbpp"
        data-orientation="horizontal"
        role="group"
      >
        <button
          class="chakra-button css-1jivwxj"
          type="button"
        >
          Buy now
        </button>
        <button
          class="chakra-button css-ox7e31"
          type="button"
        >
          Add to cart
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
